### ISSUE TITLE
The same 3D scene 3.8.3 takes up 90MB more memory than Unity 2021

### ISSUE THREAD

**Issue by jordiwang** [cocos/cocos-engine#17039]
### Use Case

Use default settings for everything, The same 3D scene 3.8.3 takes up more memory than Unity

### Problem Description

<img width="1222" alt="image" src="https://github.com/cocos/cocos-engine/assets/29055881/aa819f95-ca7b-4abd-aac0-07ce0a30ddcc">

### How it works

https://share.weiyun.com/14O8zNAF

### Alternatives Considered

Can analyze and compare the specific differences in memory usage?

**Comment by minggo** [comment]
Did you test in release mode?

**Comment by jordiwang** [comment]
> Did you test in release mode?

yes, build android release apk, code is here https://share.weiyun.com/14O8zNAF

**Comment by minggo** [comment]
May be you can try to improve it by:
- use opengl es instead of vulkan
- disable multi-thread rendering in GFXDeviceManager.h
- don't trigger `Allow Data Access` if the mesh is a static mesh, which means the mesh will not be used again after sending data to GPU
- don't include modules that are not needed in your game

**Comment by jordiwang** [comment]
Is there an official demo for performance comparison testing?

**Comment by minggo** [comment]
No, we will not provide demoes for this type of comparison.

**Comment by jordiwang** [comment]
> May be you can try to improve it by:
> 
> * use opengl es instead of vulkan
> * disable multi-thread rendering in GFXDeviceManager.h
> * don't trigger `Allow Data Access` if the mesh is a static mesh, which means the mesh will not be used again after sending data to GPU
> * don't include modules that are not needed in your game


<img width="946" alt="image" src="https://github.com/cocos/cocos-engine/assets/29055881/27f62404-0bde-4379-9ee4-8c51201c68a4">

After disabling multithreading and taking the above measures, there is still about 60MB more, are there any other measures?

**Comment by minggo** [comment]
Did you disable not needed features, such as physics.

**Comment by jordiwang** [comment]
<img width="924" alt="image" src="https://github.com/cocos/cocos-engine/assets/29055881/c5f7824a-7aaf-4ccc-9c5b-29b06a431184">
<img width="946" alt="image" src="https://github.com/cocos/cocos-engine/assets/29055881/31c58b46-6457-4d8a-b168-3a4f2ceaa861">
<img width="966" alt="image" src="https://github.com/cocos/cocos-engine/assets/29055881/c980bb5e-b722-4b1b-ae59-b2598d076799">

After disable not needed features， the memory is still 50MB larger than unity 2021

are there any other measures?

**Comment by smallmain** [comment]
Hey, at least the CPU won! Double! 🤩

**Comment by jordiwang** [comment]
Hahahaha That's a difficult trade-off.
